### ISSUE TITLE
XSData cant handle document order of elements

### ISSUE THREAD

**Issue by hcw70** [tefra/xsdata#262]
Given a schema like

```
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified"
    xmlns="http://test.de/order" targetNamespace="http://test.de/order">
    
    <xs:element name="order" type="TOrder"/>

    <xs:complexType name="TOrder">
        <xs:sequence minOccurs="0" maxOccurs="unbounded">
            <xs:choice minOccurs="0" maxOccurs="unbounded">
                <xs:element name="elem1"/>
                <xs:element name="elem2"/>
            </xs:choice>
        </xs:sequence>
    </xs:complexType>
</xs:schema>

```

and an instance of
```

<?xml version="1.0" encoding="UTF-8"?>
<order xmlns="http://test.de/order"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xsi:schemaLocation="http://test.de/order file:/home/hcw/tmp/xsdata-order/order.xsd">
  <elem1/>
  <elem2/>
  <elem1/>
  <elem2/>
</order>
```
when parsing and printing via XSData reveals:
```
<?xml version='1.0' encoding='UTF-8'?>
 <order xmlns="http://test.de/order" 
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <elem1/>
   <elem1/>
   <elem2/>
   <elem2/>
 </order>

```
so the elements are re-ordererd based on their type. Document order is not preserved.

This is critical in all contexts where the order of elements in the document is important.

**Comment by hcw70** [comment]
Parsing and printing (after xsdata generator run to module "order" is done via:

```
from order import Order
from xsdata.formats.dataclass.parsers import XmlParser
from xsdata.formats.dataclass.parsers.config import ParserConfig

config = ParserConfig(fail_on_unknown_properties=True)
parser = XmlParser(config=config)
import pathlib
order = parser.from_path(pathlib.Path("instance1.xml"), Order)

from xsdata.formats.dataclass.serializers import XmlSerializer
import pprint

serializer = XmlSerializer(pretty_print=True)
pprint.pprint(serializer.render(order))

```

**Comment by tefra** [comment]
Same cause with #263, nesting container types...

**Comment by tefra** [comment]
fixed #264
